internal/configsource/envvarconfigsource/README.md

```diff
@@ -0,0 +1,44 @@
+# Environment Variable Config Source (Alpha)
+
+Use the environmental variable config source instead of direct references to
+environment variables in the config to inject YAML fragments or to have default
+values in case the selected environment variable is undefined.
```

Comment:
It's taking me a few reads to understand what this does.  Would you mind providing an example with/without the config source and demonstrating what it can do for nonscalar yaml types (maybe with a real component)?  It the main benefit is a default value, I'd wonder why we aren't adding `"${ENV_VAR:-defaultVal}"` support to the config in general without needing a new source.

Reply:
This is about supporting the same capabilities described at https://docs.signalfx.com/en/latest/integrations/agent/remote-config.html#environment-variables

The main point is about injecting YAML fragments that is what is really missing from the current support for env vars in the collector configuration. Adding support to `${ENV_VAR:-defaultVal}` needs to be done in coordination with core but it seems a very natural idea that keeps popping up.

Anyway, if the reading it is not clear to you it is a sign that need some re-work :) I will try to improve it.

Reply:
Sorry, rereading my comment I see it wasn't clear that I picked up on the helpful functionality added, but that the readme was a little brief on the details and full capabilities.  Thanks!

Reply:
@rmfitzpatrick take a second read, please. The comment was actually quite good: it made me realize that I hadn't merged the change to support YAML fragments 🤦 that you saw via https://github.com/signalfx/splunk-otel-collector/pull/362